src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7420,7 +7420,38 @@ public StatementSyntax ParseStatement()
         }
 
         private StatementSyntax ParsePossiblyAttributedStatement()
-            => ParseStatementCore(ParseAttributeDeclarations(inExpressionContext: true), isGlobal: false);
+            => ParseStatementCore(ParseStatementAttributeDeclarations(), isGlobal: false);
+
+        private SyntaxList<AttributeListSyntax> ParseStatementAttributeDeclarations()
+        {
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return default;
+
+            // See if we should treat this as a collection expression.  At the top-level or statement-level, this should
+            // only be considered a collection if followed by a `.`, `?` or `!` (indicating it's a value, not an
+            // attribute).
```

Comment:
Right.  It likely will need special IDE support here to do well given this ambiguity.  But this is *very* normal for the IDE>  For example, we have to do all sort of special work inside of `(...` since it could be casts, parenthesized-exprs, lambdas, etc.   PITA, but in our wheelhouse :)